### PR TITLE
 Allow users to specify Trame Jupyter Mode in environment variable

### PR DESCRIPTION
**Pull request by annehaley**
### Overview

In environments such as Binder builds, it would be helpful to be able to specify whether to use the `trame_jupyter_extension` with an environment variable. This would reduce the need to reset the configuration manually every time the kernel starts. With these changes, a user may specify `PYVISTA_TRAME_JUPYTER_MODE` in the environment to override the default behavior. If set, accepted values include "extension", "proxy", and "native".

See https://mybinder.org/v2/gh/pyvista/pyvista-xarray/fix-binder?labpath=examples. This Binder build needs to have the `trame_jupyter_extension` enabled. Without it, the plotter iFrames show 404 errors.

### Details

This PR involves one change: the addition of a logic block in the constructor for `pyvista/plotting/themes/_TrameConfig`. If `PYVISTA_TRAME_JUPYTER_MODE` is set, the value will determine how to override the values for `self._server_proxy_enabled` and `self._jupyter_extension_enabled`. If not set, the default values for these are used.

The following table shows how each accepted value affects these attributes, as well as the precondition that must be true for the override to occur.

| PYVISTA_TRAME_JUPYTER_MODE | condition | server_proxy_enabled| jupyter_extension_enabled |
|-----|-----|-----------|------|
|"extension"|extension must be available|False|True|
|"proxy"|proxy must be available|True|False|
|"native"|None|False|False|

### CODE REVIEW

File: pyvista/plotting/themes.py [hunk 1]
```diff
@@ -1396,6 +1396,20 @@ def __init__(self):
...
+            if jupyter_mode == "extension" and self._jupyter_extension_available:
+                self._server_proxy_enabled = False
+                self._jupyter_extension_enabled = True
+
+            if jupyter_mode == "proxy" and self._server_proxy_enabled:
+                self._jupyter_extension_enabled = False
+
+            if jupyter_mode == "native":
+                self._jupyter_extension_enabled = False
+                self._server_proxy_enabled = False
```
Comment (tkoyama010) [inline pyvista/plotting/themes.py:1412]:
```suggestion
        # if set, jupyter_mode overwrites defaults
        jupyter_mode = os.environ.get("PYVISTA_TRAME_JUPYTER_MODE")
        if jupyter_mode == "extension" and self._jupyter_extension_available:  # pragma: no cover
            self._server_proxy_enabled = False
            self._jupyter_extension_enabled = True
        elif jupyter_mode == "proxy" and self._server_proxy_enabled:  # pragma: no cover
            self._jupyter_extension_enabled = False
        elif jupyter_mode == "native":  # pragma: no cover
            self._jupyter_extension_enabled = False
            self._server_proxy_enabled = False
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]
Thanks for this. Could you add a document about this to `doc/source/user-guide/jupyter/trame.rst`?

**Comment by annehaley** [comment]
Sure, I added a section to that page in [29e6d75](https://github.com/pyvista/pyvista/pull/5798/commits/29e6d7538daa22bd823ad3bf63b97aa7d79d02a8). Feel free to make any changes to ensure the instructions are clear. Thank you for fixing the coverage failure, too.

File: doc/source/user-guide/jupyter/trame.rst [hunk 2]
```diff
@@ -150,6 +150,51 @@ setting the following flag to ``True`` or ``False``:
...
+it may be more efficient to configure the Remote Jupyter Host with an environment variable.
+If set, the value for `PYVISTA_TRAME_JUPYTER_MODE` will determine the values of
+these two flags:
+
+* :py:attr:`pyvista.global_theme.trame.server_proxy_enabled
+  <pyvista.plotting.themes._TrameConfig.server_proxy_enabled>`
+* :py:attr:`pyvista.global_theme.trame.jupyter_extension_enabled
+  <pyvista.plotting.themes._TrameConfig.jupyter_extension_enabled>`
+
+If set, the accepted values for `PYVISTA_TRAME_JUPYTER_MODE` include "extension", "proxy", and "native".
```
Comment (tkoyama010) [inline doc/source/user-guide/jupyter/trame.rst:166]:
```suggestion
If set, the accepted values for `PYVISTA_TRAME_JUPYTER_MODE` include ``'extension'``, ``'proxy'``, and ``'native'``.
```

**Comment by tkoyama010** [comment]
@pyvista-bot preview

**Review by tkoyama010 — APPROVED** [review]
LGTM
Screenshot_20240322-081951

**Pull request merged**